### ISSUE TITLE
Router CE rename clashes with router keyword in application CE

### ISSUE THREAD

**Issue by NinoFloris** [SaturnFramework/Saturn#87]
Oops...
<img width="275" alt="screen shot 2018-07-05 at 20 10 00" src="https://user-images.githubusercontent.com/4218809/42340399-356446ca-8090-11e8-9762-dfab04b96578.png">
>'router' is not used correctly. This is a custom operation in this query or computation expression.

**Comment by Krzysztof-Cieslak** [comment]
Related to https://github.com/SaturnFramework/Saturn/issues/83

CC: @forki @isaacabraham @rmunn

**Comment by NinoFloris** [comment]
deprecate router in application and call it routes?

**Comment by rmunn** [comment]
The `router` operation is required in an application, right? So call it `primaryRouter` or `mainRouter` or `topLevelRouter` or `topRouter` or `appRouter`. I like `mainRouter` or `appRouter` the best.

**Comment by rmunn** [comment]
Or, wait, Saturn style is snake_case identifiers for CE operations, so make that `main_router` and `app_router`.

**Comment by isaacabraham** [comment]
`app_router` sounds good to me.

**Comment by isaacabraham** [comment]
Although I'm still not a huge fan of the `_` separators :-)

**Comment by NinoFloris** [comment]
I actually like routes for its simplicity "you supply a router for your application routes"

**Comment by rmunn** [comment]
Let's use #89 for discussion of snake_case vs camelCase, and use this issue for discussing what the `router` name will be renamed to. (Though it sounds like consensus has settled on `appRouter` or `app_router`, depending on the outcome of #89, so this issue may not need any more discussion.)

**Comment by Krzysztof-Cieslak** [comment]
`app_router` sounds good to me, but yeah, let's figure out naming convention first.

**Comment by NinoFloris** [comment]
@rmunn @Krzysztof-Cieslak (doh, yes I automatically wrote it F# style ;) but indeed, #89 was born out of this discussion)

About `app_router/appRouter` to me it doesn't sound like the critical and only required op that it should sound like, I actually fear the discoverability is bad with `app_router`. 

As a devils advocate question then: Why not stick with the convention the app CE employs for many others, `use_router`?

And another, if it's just a handler, why not call the op `start`/'main' or something like that and drop the router connotation?
```fs
application {
    [...]
    start (router { forward "/api" apiRouter })
}
```

P.S. This whole discussion may be moot when I write my proposal for splitting application into a config and application piece, router might move into a function arg then. (this is actually really important to get right before 1.0 because the app CE currently doesn't compose, which means you cannot easily change pieces of services/pipelines based on e.g. AspNet Environment without dropping down to app_config/service_config or switching out the full app CE)

**Comment by Krzysztof-Cieslak** [comment]
https://github.com/SaturnFramework/Saturn/commit/add78f4fc442c5606de23257694ecf4b082a2ff4 solves this particular problem by adding `use_router` operation and marking `router` (in application CE) obsolete. This change will be released as part of incoming 0.7 release. The naming convention change, if it will happen, will be part of vNext release
